### PR TITLE
[Storage] Fix issues with wrong error message

### PR DESCRIPTION
**Pull request by Juliehzl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/storage/__init__.py [hunk 1]
```diff
@@ -185,6 +185,13 @@ def handler(ex):
                 """
                 logger.error(message)
                 return
+            if isinstance(ex, t_error) and ex.status_code == 403 and ex.error_code == 'AuthorizationFailure':
+                message = """
+You maybe not allowed by the network rules of storage account. Please check network rule set using 'az storage account show -n accountname --query networkRuleSet'.
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/storage/__init__.py]: allowed to access storage account by ... ?

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -59,6 +59,7 @@ Release History
 **Storage**
 
 * storage account create: Add --enable-hierarchical-namespace to support filesystem semantics in blob service.
+* Fix issues with wrong error message "You do not have the required permissions needed to perform this operation."
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst]: Is there an issue related to this PR?

File: src/azure-cli/azure/cli/command_modules/storage/__init__.py [hunk 3]
```diff
@@ -172,7 +172,7 @@ def handler(ex):
             t_error = get_sdk(self.command_loader.cli_ctx,
                               ResourceType.DATA_STORAGE,
                               'common._error#AzureHttpError')
-            if isinstance(ex, t_error) and ex.status_code == 403:
+            if isinstance(ex, t_error) and ex.status_code == 403 and ex.error_code == 'AuthorizationPermissionMismatch':
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/storage/__init__.py]: >AuthorizationPermissionMismatch [](start = 87, length = 31)

How do you know which error_code should use? Is there a document about the error codes and their meaning?

File: src/azure-cli/HISTORY.rst (hunk 2 above)
Comment (jiasli) [reply src/azure-cli/HISTORY.rst]: This sounds better to me:
> Fix incorrect error message "You do not have the required permissions needed to perform this operation." when blocked by network rules.

File: src/azure-cli/azure/cli/command_modules/storage/__init__.py (hunk 3 above)
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/storage/__init__.py]: Upon narrowing `ex.error_code` to `AuthorizationPermissionMismatch` and `AuthorizationFailure`, if the key or connection string is incorrect, `ex.error_code` will be `AuthenticationFailed` which is not handled.

File: src/azure-cli/azure/cli/command_modules/storage/__init__.py (hunk 1 above)
Comment (jiasli) [reply src/azure-cli/azure/cli/command_modules/storage/__init__.py]: How about
> The request may be blocked by network rules of the storage account.

**Pull request closed without merging**